### PR TITLE
HBASE-25285 [branch-1] Fix two methods removed problems reported by compat-check

### PR DESCRIPTION
**Pull request by Reidddddd**

### CODE REVIEW

**Comment by Reidddddd** [comment]
Because they are HBaseInterfaceAudience.COPROC.
Let's keep them, just in case any coprocessor uses.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m  0s |  Docker mode activated.  |
| -1 :x: |  docker  |  10m 40s |  Docker failed to build yetus/hbase:b249092a5f.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hbase/pull/2658 |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2658/1/console |
| versions | git=2.17.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m  0s |  Docker mode activated.  |
| -1 :x: |  docker  |   0m 11s |  Docker failed to build yetus/hbase:b249092a5f.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hbase/pull/2658 |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2658/2/console |
| versions | git=2.17.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Reidddddd** [comment]
```
Archive JUnit-formatted test results
[2020-11-16T07:23:14.835Z] Recording test results
[2020-11-16T07:23:14.982Z] No test report files were found. Configuration error?
```
any hint?

File: hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServer.java [hunk 1]
```diff
@@ -2357,6 +2357,11 @@ public synchronized void start() {
     started = true;
   }
 
+  @Override
+  public void refreshAuthManager(PolicyProvider pp) {
+    // noop
```
Comment (apurtell) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServer.java:2362]: Can we not call the static method from here?

**Review by apurtell — CHANGES_REQUESTED** [review]
One change requested, otherwise lgtm

File: hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServer.java (hunk 1 above)
Comment (Reidddddd) [reply hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServer.java:2362]: `RpcServer` is IA.private, and #refreshAuthManager shouldn't be called elsewhere, do we still need the concrete impl

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m  0s |  Docker mode activated.  |
| -1 :x: |  docker  |   0m 12s |  Docker failed to build yetus/hbase:b249092a5f.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hbase/pull/2658 |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2658/3/console |
| versions | git=2.17.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServerInterface.java [hunk 2]
```diff
@@ -78,6 +78,15 @@
    */
   void addCallSize(long diff);
 
+  /**
+   * It was removed in HBASE-24174 and added back in HBASE-25285 for compactibility concern.
+   * NOTICE: implementations should call {@link #refreshAuthManager(Configuration, PolicyProvider)}
+   * for correctness.
+   * @param pp PolicyProvider
+   */
+  @VisibleForTesting
```
Comment (apurtell) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServerInterface.java:87]: Add a Deprecated annotation. Can be done at commit time.

**Review by apurtell — APPROVED** [review]
Consider adding a Deprecated annotation to all of these compat methods we put back. Not a big deal. Can be done at commit time.

**Review by virajjasani — APPROVED** [review]
+1

**Comment by virajjasani** [comment]
Triggered new build, we should see QA result this time.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |  12m  0s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ branch-1 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   9m 53s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 40s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  compile  |   0m 45s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  checkstyle  |   1m 42s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m  2s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 48s |  branch-1 passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   0m 41s |  branch-1 passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +0 :ok: |  spotbugs  |   3m  8s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   3m  4s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m 58s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 41s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javac  |   0m 41s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 45s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  javac  |   0m 45s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   1m 32s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   2m 49s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   4m 38s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 30s |  the patch passed with JDK Azul Systems, Inc.-1.8.0_262-b19  |
| +1 :green_heart: |  javadoc  |   0m 41s |  the patch passed with JDK Azul Systems, Inc.-1.7.0_272-b10  |
| +1 :green_heart: |  findbugs  |   2m 54s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  | 125m 25s |  hbase-server in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 35s |  The patch does not generate ASF License warnings.  |
|  |   | 178m 33s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2658/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2658 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux db8abcbc8621 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-home/workspace/Base-PreCommit-GitHub-PR_PR-2658/out/precommit/personality/provided.sh |
| git revision | branch-1 / edccfe4 |
| Default Java | Azul Systems, Inc.-1.7.0_272-b10 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:Azul Systems, Inc.-1.8.0_262-b19 /usr/lib/jvm/zulu-7-amd64:Azul Systems, Inc.-1.7.0_272-b10 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2658/4/testReport/ |
| Max. process+thread count | 3352 (vs. ulimit of 10000) |
| modules | C: hbase-server U: hbase-server |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2658/4/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   7m 52s |  Docker mode activated.  |
| -1 :x: |  patch  |   0m  8s |  https://github.com/apache/hbase/pull/2658 does not apply to branch-1. Rebase required? Wrong Branch? See https://yetus.apache.org/documentation/in-progress/precommit-patchnames for help.  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2658/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2658 |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2658/5/console |
| versions | git=1.9.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Pull request merged**